lib/membrane/video_compositor/examples/mock/layouts/position.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Membrane.VideoCompositor.Examples.Mock.Layouts.Position do
+  @moduledoc """
+  A structure representing position of an object in rendered frame.
+  """
+
+  @enforce_keys [:top_left_corner, :width, :height]
+  defstruct @enforce_keys ++ [z_value: 0.0]
+
+  @typedoc """
+  Float in [0, 1] range. Widely used in graphic rendering, to determine distance, color etc.
+  """
+  @type unit_range :: float()
+
+  @type t :: %__MODULE__{
+          top_left_corner: {unit_range(), unit_range()},
+          width: unit_range() | :auto,
+          height: unit_range() | :auto,
+          z_value: unit_range()
```

Comment:
Why are these floats? It seems quite unusual to define either of these as floats. Additionally, what do the `width` and `height` refer to? The scale of the input or percentage of available width or height? It seems ambiguous

Reply:
Since input texture resolution (input from some membrane pad) might change, depending on e.g. network conditions, defining cropping parameters in relative coordinates (like [0, 1] range used here) can deal with such issues.